homeassistant/components/anova/config_flow.py

```diff
@@ -0,0 +1,60 @@
+"""Config flow for Anova."""
+from __future__ import annotations
+
+from anova_wifi import AnovaApi, InvalidLogin, NoDevicesFound
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers import aiohttp_client
+
+from .const import DOMAIN
+
+
+class AnovaConfligFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Sets up a config flow for Anova."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, str] | None = None
+    ) -> FlowResult:
+        """Handle a flow initiated by the user."""
+        errors: dict[str, str] = {}
+        if user_input is not None:
+            api = AnovaApi(
+                aiohttp_client.async_get_clientsession(self.hass),
+                user_input[CONF_USERNAME],
+                user_input[CONF_PASSWORD],
+            )
+            await self.async_set_unique_id(user_input[CONF_USERNAME].lower())
+            self._abort_if_unique_id_configured()
+            try:
+                await api.authenticate()
+                devices = await api.get_devices()
+            except InvalidLogin:
+                errors["base"] = "invalid_auth"
+            except NoDevicesFound:
+                errors["base"] = "no_devices_found"
+            except Exception:  # pylint: disable=broad-except
+                errors["base"] = "unknown"
+            else:
+                # We store device list in config flow in order to persist found devices on restart, as the Anova api get_devices does not return any devices that are offline.
+                device_list = [(device.device_key, device.type) for device in devices]
```

Comment:
This code appears in `__init__.py` so it could be a function to build the device list

Reply:
What do you mean by that? Have a util function to take a list of devices and turn it into this serializable form?

Reply:
That is what I am assuming so I moved it into utils. Please correct me if this is not what you meant.